pkg/policy/api/rule.go

```diff
@@ -590,3 +633,31 @@ func (kr *PortRuleKafka) GetAPIVersion() (int16, bool) {
 
 	return *kr.apiVersionInt, false
 }
+
+// MapRoleToAPIKey maps the Role to the low level set of APIKeys for that role
+func (kr *PortRuleKafka) MapRoleToAPIKey() bool {
+	if len(kr.Role) <= 0 {
+		return false
+	}
+
+	if len(kr.apiKeyInt) > 0 {
+		return true
+	}
+
+	// Expand the kr.apiKeyInt array based on the Role.
+	// For produce role, we need to add mandatory apiKeys produce, metadata and
+	// apiversions. While for consume, we need to add mandatory apiKeys like
+	// fetch, offsets, offsetcommit, offsetfetch, apiversions, metadata,
+	// findcoordinator, joingroup, heartbeat,
+	// leavegroup and syncgroup.
+	switch strings.ToLower(kr.Role) {
+	case "produce":
```

Comment:
Please define this and `"consume"` as constants in `pkg/policy/api/rule.go`

Reply:
sure

Reply:
done